api2-war/src/main/java/eu/europeana/api2/v2/service/translate/TranslationUtils.java

```diff
@@ -228,4 +237,62 @@ public static String truncateFieldValue(String fieldValue, Integer truncateAfter
         }
         return WordUtils.abbreviate(fieldValue, truncateAfter, truncateHardLimit, TRUNCATED_INDICATOR);
     }
+
+    public static List<String> truncate(List<String> valuesToTranslate , Integer translationCharLimit, Integer translationCharTolerance) {
+        List<String> truncatedValues = new ArrayList<>();
+        if (truncationNotRequired(valuesToTranslate, translationCharLimit)) {
+            return valuesToTranslate;
+        }
+        boolean noFurtherLooking = false;
+        Integer charAccumulated = 0;
+        for (String value : valuesToTranslate) {
+            // check if the value exceeded the limit.
+            if ((charAccumulated + value.length()) >= translationCharLimit) {
+                // get exceeded String value
+                Integer charLimitIndex = translationCharLimit - charAccumulated;
+                String valueAfterLimit = StringUtils.substring(value, charLimitIndex, value.length());
+
+                //  check if the string has a phrase or new line
+                if (hasPhraseOrNewLine(valueAfterLimit)) {
+                    Matcher m = getValuesBeforePhraseOrNewLinePattern.matcher(valueAfterLimit);
+                    if (m.find()) {
+                           truncatedValues.add(StringUtils.substring(value, 0,  charLimitIndex) + m.group(0) + TRUNCATED_INDICATOR) ;
```

Comment:
Doesn't getValuesBeforePhraseOrNewLinePattern return an empty string if no phrase or new line is found? If so we could simply skip the hasPhraseOrNewLine method and instead do getValuesBeforePhraseOrNewLine and check return value?

Reply:
yes, we can directly use m.find() and get the first group.

Reply:
it seems the getValuesBeforePhraseOrNewLinePattern if the phrase or new line is not present , will match the whole string.

Reply:
have updated the pattern now, if there is no phrase or new line. The string will empty